src/main/java/com/google/sps/data/DatastoreClient.java

```diff
@@ -0,0 +1,136 @@
+import com.google.sps.data.Session;
+import com.google.sps.data.SessionInterface;
+import com.google.sps.data.Instance;
+import com.google.sps.data.InstanceInterface;
+
+/** Class that reads from and writes to Datastore. */
+public class DatastoreClient implements DatastoreClientInterface {
+  /** 
+   * Adds an attendeeInterface object to the Datastore. If object already
+   * exists, the old one is replaced. 
+   */
+  public void insertOrUpdateAttendee(AttendeeInterface attendee) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(attendee.toEntity());
+  }
+
+  /** 
+   * Adds an InstanceInterface object to the Datastore. If object already
+   * exists, the old one is replaced. 
+   */
+  public void insertOrUpdateInstance(InstanceInterface instance) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(instance.toEntity());
+  }
+
+  /** 
+   * Adds an SessionInterface object to the Datastore. If object already
+   * exists, the old one is replaced. 
+   */
+  public void insertOrUpdateSession(SessionInterface session) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(session.toEntity());
+  }
+
+  /** 
+   * Returns the Session object associated with given sessionId. If 
+   * return Optional is empty, then no object exists with given parameter.
+   */
+  public Optional<Session> getSession(String sessionId) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query getSession = new Query(EntityConstants.SessionEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate(EntityConstants.SessionEntity.SESSION_ID,
+      FilterOperator.EQUAL, sessionId));
+    PreparedQuery pq = datastore.prepare(getSession);
+    Entity sessionEntity = pq.asSingleEntity();
+    return Optional.of(Session.fromEntity(sessionEntity));
+  }
+
+   /** 
+    * Returns the Instance object associated with given instanceName. If 
+    * return Optional is empty, then no object exists with given parameter. 
+    */  
+  public Optional<Instance> getInstance(String instanceName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query getInstance = new Query(EntityConstants.InstanceEntity.TABLE_NAME)
+      .setFilter(new FilterPredicate
+      (EntityConstants.InstanceEntity.INSTANCE_NAME, 
+      FilterOperator.EQUAL, instanceName));
+    PreparedQuery pq = datastore.prepare(getInstance);
+    Entity instanceEntity = pq.asSingleEntity();
+    return Optional.of(Instance.fromEntity(instanceEntity));
+   }
+
+  /** 
+   * Returns the Attendee object associated with given screenName. If 
+   * return Optional is empty, then no object exists with given parameter.
+   */  
+  public Optional<Attendee> getAttendee(String screenName) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
You dont need to instantiate this in every single function. Just make a private instance variable. And also please change its name from datastore to datastoreService